eole/models/model.py

```diff
@@ -892,7 +898,17 @@ def forward(self, src, tgt, src_len, bptt=False, with_align=False):
             with_align=with_align,
             tgt_pad_mask=pad_mask,
         )
-        return dec_out, attns, None
+
+        if self.add_estimator:  # we take the average of dec_out using the pad mask
+            pad_mask2 = ~src.eq(pad_idx)
+            in_estim2 = (dec_out * pad_mask2.unsqueeze(-1).float()).sum(
+                dim=1
+            ) / pad_mask2.sum(dim=1, keepdim=True).float()
+            estim = self.estimator(in_estim2.to(dec_out.dtype)).squeeze(-1)
+        else:
+            estim = None
+
+        return dec_out, attns, estim
```

Comment:
Duplicated code. Can't we factorize in some `apply_estimator` method?

Reply:
yes agreed, but ideally it would best to handle the non averaged case for encoder only models, so not very easy